extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/IndexerControllerContext.java

```diff
@@ -269,6 +270,14 @@ public static Map<String, Object> makeTaskContext(
         .put(MultiStageQueryContext.CTX_IS_REINDEX, MSQControllerTask.isReplaceInputDataSourceTask(querySpec))
         .put(MultiStageQueryContext.CTX_MAX_CONCURRENT_STAGES, queryKernelConfig.getMaxConcurrentStages());
 
+    // Put the lookup names in the query context to facilitate selective loading of lookups.
+    if (querySpec.getQuery().getContext().containsKey(PlannerContext.CTX_LOOKUPS_TO_LOAD) && querySpec.getQuery().getContext().get(PlannerContext.CTX_LOOKUPS_TO_LOAD) != null) {
```

Comment:
The first condition seems redundant:
```suggestion
    if (querySpec.getQuery().getContext().get(PlannerContext.CTX_LOOKUPS_TO_LOAD) != null) {
```